### PR TITLE
init

### PR DESCRIPTION
**Pull request by njuettner**
First draft:

Tagger will run as a cronjob and tags all volume id's inside the WC.

With the first version, customer will provide a configmap which is getting collected by the tagger but only keys starting with `tag.provider.giantswarm.io` everything else will be ignored. This is to ensure not overriding any other tags.

Configmap is the source of truth, if tags are not in sync it will remove tags  starting with `tag.provider.giantswarm.io` from EBS volumes.

Be aware the configmap name is called `test` but I will change it immediately after testing it properly.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM alpine:3.13.5
```
Comment (whites11) [inline Dockerfile]: this is very old, can we use a newer alpine? latest is 3.16.2

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,17 @@
+module github.com/giantswarm/aws-ebs-volume-tagger
+
+go 1.16
```
Comment (whites11) [inline go.mod]: also very old, can we use 1.18 at least?

File: helm/aws-ebs-volume-tagger/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v2
+appVersion: 0.0.1
```
Comment (whites11) [inline helm/aws-ebs-volume-tagger/Chart.yaml]:
```suggestion
appVersion: 0.1.0
```

Comment (whites11) [reply helm/aws-ebs-volume-tagger/Chart.yaml]: First valid SEMVER version is 0.1.0

File: helm/aws-ebs-volume-tagger/templates/cronjob.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ .Release.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+spec:
+  concurrencyPolicy: Allow
```
Comment (whites11) [inline helm/aws-ebs-volume-tagger/templates/cronjob.yaml]: I'm not sure about having concurrency allowed. There might be some race condition I think. WDYT?

File: helm/aws-ebs-volume-tagger/values.yaml [hunk 5]
```diff
@@ -7,6 +7,8 @@ project:
 
 image:
   registry: quay.io
-  name: giantswarm/{APP-NAME}
+  name: giantswarm/aws-ebs-volume-tagger
   tag: v0.0.1
```
Comment (whites11) [inline helm/aws-ebs-volume-tagger/values.yaml]: I think you need to use `[[ .Version ]]` here in order for automation to use the right version based on the release or the SHA of the PR.

File: helm/aws-ebs-volume-tagger/templates/cronjob.yaml (hunk 4 above)
Comment (njuettner) [reply helm/aws-ebs-volume-tagger/templates/cronjob.yaml]: good idea I will drop it 👍🏻

File: Dockerfile (hunk 1 above)
Comment (njuettner) [reply Dockerfile]: probably the github template needs to be updated. I didn't look at it, will update thanks <3

File: go.mod (hunk 2 above)
Comment (njuettner) [reply go.mod]: yeah will do, the gh template seems to be really outdated and I didn't check lets update to at least 1.18 💪🏻

File: helm/aws-ebs-volume-tagger/values.yaml (hunk 5 above)
Comment (njuettner) [reply helm/aws-ebs-volume-tagger/values.yaml]: good point, will do that 👍🏻

File: helm/aws-ebs-volume-tagger/templates/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    - "persistentvolumes"
+    verbs: 
+    - "get"
+    - "list"
+    - "watch"
+  - apiGroups: 
+    - ""
+    resources: 
+    - "configmaps"
+    verbs:
```
Comment (whites11) [inline helm/aws-ebs-volume-tagger/templates/rbac.yaml:18]: Apparently you don't need all these permissions. just `list` for persistentvolumes` and `get` for configmaps.
You could also restrict the configmap get to a specific name once you're settled with the `test` name change

**Review by whites11 — APPROVED** [review]
seems reasonable

File: helm/aws-ebs-volume-tagger/templates/rbac.yaml (hunk 6 above)
Comment (njuettner) [reply helm/aws-ebs-volume-tagger/templates/rbac.yaml:18]: you're right, this is a good idea ❤️ .

**Pull request merged**